src/eddymotion/model/dipy.py

```diff
@@ -348,12 +392,61 @@ def compute_spherical_covariance(
 
     _ensure_positive_scale(a)
 
-    return np.where(theta <= a, 1 - 3 * (theta / a) ** 2 + 2 * (theta / a) ** 3, 0)
+    return np.where(theta <= a, 1 - 3 * theta / (2 * a) + theta**3 / (2 * a**3), 0)
+
+
+def compute_derivative(
+    theta: np.ndarray,
+    kernel: np.ndarray,
+    weighting: str,
+    params: dict[float],
+):
+    """
+    Compute the analytical derivative of the kernel.
+
+    Parameters
+    ----------
+    theta : :obj:`~numpy.ndarray`
+        Pairwise angles across diffusion gradient encoding directions.
+    kernel : :obj:`~numpy.ndarray`
+        Current kernel.
+    weighting : :obj:`str`
+        The kind of kernel which derivatives will be calculated.
+    params : :obj:`dict`
+        Current parameter values.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Gradients of the kernel.
+
+    """
+
+    n_partials = len(params)
+
+    a = params.pop("a")
+    lambda_s = params.pop("lambda_s")
+
+    min_angles = theta > a
+
+    if weighting == "spherical":
+        deriv_a = 1.5 * (theta[min_angles] / a**2 - theta[min_angles] ** 3 / a**4)
```

Comment:
Should be $$\frac{3\theta}{2a^2} - \frac{3\theta^3}{2a^4}$$ So missing a 1.5 multiplier to the last term.


```suggestion
        deriv_a = 1.5 * (theta[min_angles] / a**2 - 1.5 * theta[min_angles] ** 3 / a**4)
```

Reply:
> Should be $$\frac{3\theta}{2a^2} - \frac{3\theta^3}{2a^4}$$ So missing a 1.5 multiplier to the last term.

The current implementation is $$\frac{3}{2}\left( \frac{\theta}{a^2} - \frac{\theta^3}{a^4} \right)$$, please note the parentheses around the fractional sum in the code.